lib/WebpackConfig.js

```diff
@@ -346,27 +346,32 @@ class WebpackConfig {
     }
 
     addEntry(name, src) {
-        if (this.entries.has(name)) {
-            throw new Error(`Duplicate name "${name}" passed to addEntry(): entries must be unique.`);
-        }
-
-        // also check for styleEntries duplicates
-        if (this.styleEntries.has(name)) {
-            throw new Error(`The "${name}" passed to addEntry conflicts with a name passed to addStyleEntry(). The entry names between addEntry() and addStyleEntry() must be unique.`);
-        }
+        this.validateNameIsNewEntry(name);
 
         this.entries.set(name, src);
     }
 
-    addStyleEntry(name, src) {
-        if (this.styleEntries.has(name)) {
-            throw new Error(`Duplicate name "${name}" passed to addStyleEntry(): entries must be unique.`);
+    /**
+     * Provide a has of entries at once, as an alternative to calling `addEntry` several times.
+     *
+     * @param {Object.<string, string|array>} entries
+     * @returns {Void}
+     */
+    addEntries(entries = {}) {
+        if (typeof entries !== 'object') {
+            throw new Error('Argument 1 to addEntries() must be an object.');
         }
 
-        // also check for entries duplicates
-        if (this.entries.has(name)) {
-            throw new Error(`The "${name}" passed to addStyleEntry() conflicts with a name passed to addEntry(). The entry names between addEntry() and addStyleEntry() must be unique.`);
-        }
+        Object.keys(entries).forEach((key) => this.validateNameIsNewEntry(key));
+
+        this.entries = new Map([
+            ...this.entries,
+            ...Object.entries(entries),
+        ]);
```

Comment:
To me it would have been simpler to: 
- not update `addEntry` method
- don't introduce a new `validateNameIsNewEntry` method
- iterate over `entries` and call `this.addEntry()` 

WDYT?

Reply:
Extracting out logic into `validateNameIsNewEntry` still makes sense, for single responsibility.

Regarding whether `addEntries` should call `addEntry` in a loop, to set the data, or do a merge directly...
- by calling `addEntry` in a loop, a single method is responsible for updating `this.entries`, which is good

Reply:
@Kocal I've pushed up another commit, taking into some of your recommendations. Please resolve this thread, if you find the latest changes meet your concerns.

Reply:
It looks good to me, unfortunately I don't have the butto, "Resolve this thread" :grimacing: (maybe because my comment was not a review comment)